cmd/ltagent/init.go

```diff
@@ -126,9 +129,95 @@ func RunInitCmdF(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("error while initializing loadtest: %w", err)
 	}
 
+	err = genAdmins(config, userPrefix)
+	if err != nil {
+		return fmt.Errorf("error while generating admin users: %w", err)
+	}
+	mlog.Info("admin generation completed")
+
 	return genData(lt, 50)
 }
 
+func genAdmins(config *loadtest.Config, userPrefix string) error {
+	mlog.Info(fmt.Sprintf("generating %d admins", config.InstanceConfiguration.NumAdmins))
+
+	adminStore, err := memstore.New(nil)
+	if err != nil {
+		return err
+	}
+	adminUeSetup := userentity.Setup{
+		Store: adminStore,
+	}
+	adminUeConfig := userentity.Config{
+		ServerURL:    config.ConnectionConfiguration.ServerURL,
+		WebSocketURL: config.ConnectionConfiguration.WebSocketURL,
+		Username:     "",
+		Email:        config.ConnectionConfiguration.AdminEmail,
+		Password:     config.ConnectionConfiguration.AdminPassword,
+	}
+	sysadmin := userentity.New(adminUeSetup, adminUeConfig)
+
+	for i := 0; i < int(config.InstanceConfiguration.NumAdmins); i++ {
+		userStore, err := memstore.New(nil)
+		if err != nil {
+			return err
+		}
+		user := &model.User{
+			Password: "testPass123$",
+			Email:    fmt.Sprintf("%s-%d@example.com", userPrefix, i),
+			Username: fmt.Sprintf("%s-%d", userPrefix, i),
+		}
+		ueConfig := userentity.Config{
+			ServerURL:    config.ConnectionConfiguration.ServerURL,
+			WebSocketURL: config.ConnectionConfiguration.WebSocketURL,
+			Username:     user.Username,
+			Email:        user.Email,
+			Password:     user.Password,
+		}
+
+		userId, err := createUser(sysadmin, user)
+		if err != nil {
+			return err
+		}
+
+		user.Id = userId
+		err = userStore.SetUser(user)
+		if err != nil {
+			return err
+		}
+		userSetup := userentity.Setup{
+			Store: userStore,
+		}
+		err = loadtest.PromoteToAdmin(sysadmin, userentity.New(userSetup, ueConfig))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createUser(sysadmin *userentity.UserEntity, user *model.User) (string, error) {
```

Comment:
Is this needed? The init should exit in the presence of conflicts and otherwise be run on an empty state. Were you having some issues on re-runs?

Reply:
oh i see, i thought that there can be a scenario that we run it on a dirty state. Will defo remove to reduce complexityl

Reply:
The only way to do that would be to change user prefixes but it's only done for local testing. We can work on the assumption that it will always run on a fresh state.